proposals/0000-extension-lifecycle-framework.md

```diff
@@ -0,0 +1,110 @@
+
+However, once an extension clears the review process and gets implemented in the compiler, there is currently no framework for understanding or managing its evolution. The strength of the development model is its openness, allowing experimental features to be tried out in the community. Features that the community discovers to be problematic in practice can be not enabled and the community can, over time, reach a consensus about the value of these extensions.
+
+The weakness of this model is that, over time, the accumulation of programming extensions in various states of use and disuse can create confusion, especially for developers being introduced to Haskell. There is not one Haskell, but many, and each individual or team must necessarily choose their own. Not everyone has yet had the opportunity to accumulate the experience and knowledge that will allow them to quickly determine whether or not to use a given language extension, leading to various points of stress.
+
+The authors of this proposal believe that a clear lifecycle for these programming language extensions would make it easier for newcomers, those responsible for onboarding new Haskell developers as well as managers, and the community more broadly, to make sense of the Haskell language extensions. This framework would make it clear which extensions are prone to rapid evolution, which of them are stable and suitable for serious use, and which of them are generally not considered suitable for new development.
+
+## 2. Proposed Change Specification
+
+### 2.1. Categories and Lifecycle
+Language extensions will be classified into the following categories
+ * `Stable` extensions are considered to be finished and are not expected to undergo regular changes. These features can be used without worry of unexpected changes, and they are not known to contain serious design or implementation deficiencies. Any breaking change to a stable extension will be announced well in advance of the change being made, with a migration path provided if possible. For an extension to be classified as `Stable` it must be considered `Stable` when used in combination of all possible, non-mutually exclusive, extensions that are already `Stable`. This ensures any extension that is `Stable` will be have the same use without worry when combined with any other extension in the set, while allowing for mutually exclusive extensions to be added. Ideally, no breaking change will be made to a `Stable` extension, with incompatible changes resulting instead in a new, related extension, that is possibly mutually exclusive with the existing one, to enable smooth migration. Some extensions in this category might be `MultiWayIf`, `MonoLocalBinds`, and `ViewPatterns`.
```

Comment:
```suggestion
 * `Stable` extensions are considered to be finished and are not expected to undergo regular changes. These features can be used without worry of unexpected changes, and they are not known to contain serious design or implementation deficiencies.   More specifically
  * For an extension to be classified as `Stable` it must be considered `Stable` when used in combination of all possible, non-mutually exclusive, extensions that are already `Stable`. This ensures any extension that is `Stable` will be have the same use without worry when combined with any other extension in the set, while allowing for mutually exclusive extensions to be added. 
  * Ideally, no breaking change will be made to a `Stable` extension.  Instead, it may be better to add a new, related extension, that is possibly mutually exclusive with the existing one, to enable smooth migration. 
  * If, however, a breaking change is made to a stable extension, it will be announced well in advance of the change being made, with a migration path provided if possible.   
  * Some extensions in the `Stable` category might be `MultiWayIf`, `MonoLocalBinds`, and `ViewPatterns`.
```

Reply:
"This ensures any extension that is `Stable` will be have the same use without worry " doe not parse.

Reply:
Any thoughts on this?

Reply:
I thought the wording changed slightly in the last update but I must have put the existing one back before committing.
Perhaps:

> This ensures any extension that is `Stable` will have the same stability properties when combined with any other extension in the set, while allowing for mutually exclusive extensions to be added.
?